### ISSUE TITLE
Secrets Properties should not be logged

### ISSUE THREAD

**Issue by kahootali** [spring-cloud/spring-cloud-kubernetes#267]
I am using this plugin for my application to read normal data from configmap and sensitive data from secrets, but as the application loads, it is logging the secrets properties with no encryption at INFO level. It is so dangerous as logs are usually shared through different teams for debugging of an app, and one doesn't want secrets to be read by other teams.

**Comment by spencergibb** [comment]
I don't see where we log anything https://github.com/spring-cloud/spring-cloud-kubernetes/blob/master/spring-cloud-kubernetes-config/src/main/java/org/springframework/cloud/kubernetes/config/SecretsPropertySource.java

Can you post a sample of the lob (scrubbed obviously) so we can see what loggers are doing the logging?

**Comment by spencergibb** [comment]
So `SecretsPropertySource` extends `MapPropertySource` from spring framework, which, when debug is enabled (as you seem to have), prints the contents of the source. I assume that if you turn debug off, the properties are not printed?

The fix here is to override `toString()` in `SecretsPropertySource` to only log the name. Something like

```
	@Override
	public String toString() {
		return getClass().getSimpleName() + " {name='" + this.name + "'}";
	}
```

**Comment by toonborgers** [comment]
I'd like to have a go at this one

**Comment by spencergibb** [comment]
go for it, let me know if you need help.

**Comment by kahootali** [comment]
I have turned DEBUG level printing off and it doesn't show properties then, only shows the name.

**Comment by toonborgers** [comment]
@spencergibb Does this require unit tests or is that overkill? `SecretsPropertySource` currently does not have any tests

**Comment by spencergibb** [comment]
Yes, generally whenever we change functionality we want a test to prevent a regression

**Comment by toonborgers** [comment]
@spencergibb PR created
